source/agora/common/Transaction.d

```diff
@@ -90,12 +91,50 @@ public struct Transaction
 
     public void serialize (scope SerializeDg dg) pure const nothrow @safe
     {
+        serializePart(this.inputs.length, dg);
         foreach (const ref input; this.inputs)
             serializePart(input, dg);
 
+        serializePart(this.outputs.length, dg);
         foreach (const ref output; this.outputs)
             serializePart(output, dg);
     }
+
+    /***************************************************************************
+
+        Transactions deserialization
+
+        Params:
+            dg = Deserialize function
+
+    ***************************************************************************/
+
+    public void deserialize (scope DeserializeDg dg) nothrow @safe
+    {
+        ulong inputSize;
+        deserializePart(inputSize, dg);
+        this.inputs.length = inputSize;
+
+        // deserialize and generate inputs
+        foreach (idi; 0 .. inputSize)
```

Comment:
Either `idx` or `input_idx`, I don't want us to have to guess what "idi" and "ido" mean.

Reply:
modify idx